libseccomp/src/notify.rs

```diff
@@ -380,6 +448,47 @@ impl ScmpNotifResp {
 /// # Errors
 ///
 /// If the notification ID is invalid, an error will be returned.
+///
+/// # Examples
+///
+/// ```
+/// # use libseccomp::*;
+/// # use std::{fs::File, io::Read, process::Command, thread};
+/// // Get the current process ID
+/// let parent_pid = std::process::id();
+/// let notify_fd = thread::spawn(|| -> Result<i32, error::SeccompError> {
+///     let mut ctx = ScmpFilterContext::new(ScmpAction::Allow)?;
+///     ctx.add_rule(ScmpAction::Notify, ScmpSyscall::from_name("openat")?)?
+///         .load()?;
+///     let notify_fd = ctx.get_notify_fd()?;
+///
+///     /// Example command execution
+///     let _ = Command::new("ls").spawn();
+///     Ok(notify_fd)
+/// })
+/// .join()
+/// .unwrap()?;
+///
+/// let req = ScmpNotifReq::receive(notify_fd)?;
+/// assert_eq!(req.data.arch, ScmpArch::native());
+/// assert_eq!(req.data.syscall, ScmpSyscall::from_name("openat")?);
+///
+/// let mut stat = File::open(&format!("/proc/{}/stat", req.pid))?;
+///
+/// // Checks if a notification is still valid.
+/// notify_id_valid(notify_fd, req.id)?;
+///
+/// let mut stat_data = String::new();
+/// let _ = stat.read_to_string(&mut stat_data)?;
+/// println!("{}", stat_data);
+///
+/// // Checks if a notification is still valid.
+/// notify_id_valid(notify_fd, req.id)?;
+///
+/// let resp = ScmpNotifResp::new_val(req.id, parent_pid as i64, ScmpNotifRespFlags::empty());
```

Comment:
Maybe we could just use continue here. Or `new_error`. Because retuning `parent_pid` for a `open_at` call does not make sense.